structuredheadings/plugin.js

```diff
@@ -434,7 +434,45 @@
 
       return headings;
     },
-    clearAll: function (editor, element) {
+    clearAllInSelection: function (editor) {
+      var selection = editor.getSelection();
+      if (selection) {
+        var range = selection.getRanges()[0];
+        if (range.collapsed) {
+          this.clearAllFromElement(
+            editor,
+            CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)
+          );
+          return;
+        } else {
+          this.clearAllFromRange(editor, range);
+        }
+
+      }
+    },
+    clearAllFromRange: function (editor, range) {
+      var walker = new CKEDITOR.dom.walker(range); // eslint-disable-line new-cap
+      walker.evaluator = function (node) {
```

Comment:
I'd love to see this function first assigned to a variable with a good name describing what it does, then assigned to the evaluator. Reading this code, I had to dive into the function completely to figure out what was going on. My guess on the name is `is_heading_p_or_pre`